docs/architecture/stemming_and_synonyms.md

```diff
@@ -1,158 +0,0 @@
-import FeatureAvailability from '@site/src/components/FeatureAvailability';
-
-# About DataHub [Stemming and Synonyms Support]
```

Comment:
@jjoyce0510 was your intent to remove this entirely, or just to clean up the weird brackets around `[Stemming and Synonyms Support]` in the doc?

The doc looks like it's got decent content, so don't want to remove it unless it's incorrect or misleading

Reply:
https://acryldata.slack.com/archives/C01GRF0KMD5/p1691079083763849
we intent to remove it